cli/flox/src/commands/edit.rs

```diff
@@ -130,25 +165,55 @@ impl Edit {
     // instead of just environment is a pain
     async fn edit_manifest(
         flox: &Flox,
-        environment: &mut Box<dyn Environment>,
-        active_environment: UninitializedEnvironment,
-        description: String,
+        environment: &mut ConcreteEnvironment,
         contents: Option<String>,
     ) -> Result<()> {
-        match maybe_migrate_environment_to_v1(flox, environment, &description).await {
+        // TODO: we have various functionality spread across
+        // UninitializedEnvironment, ConcreteEnvironment, and
+        // Environment.
+        // UninitializedEnvironment is used to compare to what
+        // environments are active.
+        // description can't currently be derived from an Environment
+        // but is used for messages.
+        // Environment is what we'll actually use to perform the edit.
+
+        if let ConcreteEnvironment::Managed(ref environment) = environment {
+            if environment.has_local_changes(flox)? && contents.is_none() {
+                bail!(ManagedEnvironmentError::CheckoutOutOfSync)
+            }
+        };
+
```

Comment:
question nonblocking: did you mean to leave this one even though `maybe_migrate_environment_to_v1` will check as well? Not sure if that was deliberate to be explicit

Reply:
Oh that one's there to error before the interactive edit